compiler/rustc_trait_selection/src/solve/fulfill/derive_errors.rs

```diff
@@ -269,7 +269,7 @@ impl<'tcx> BestObligation<'tcx> {
         goal: &inspect::InspectGoal<'_, 'tcx>,
         self_ty: Ty<'tcx>,
     ) -> ControlFlow<PredicateObligation<'tcx>> {
-        assert!(!self.consider_ambiguities);
+        //assert!(!self.consider_ambiguities);
```

Comment:
We now assemble no candidates if the trait goal for an normalizes-to is ambiguous. 

We could wrap the ambiguity in that `let Some(proven_via) = proven_via else {` line above with a candidate kind to fix this assertion, or otherwise just remove it.